### DISCUSSION TITLE
ROS subscriber with Visp

### DISCUSSION THREAD

**Discussion by shrutichakraborty** [lagadic/visp#1333] [Q&A]
Hello, I'd like to know if its possible to write a ros2 cpp subscriber node within visp that subscribes to a topic published from ros2 and then uses these values for visual servoing? Basically I have my neural network that detects objects and returns bounding boxes in python that I can publish to a topic to read on the visp side and then perform visual servoing? If it is possible a minimal example would be very helpful. 

I have also tried to run my detector on visp but unfortunately there are errors with the onnx interpreter. Any suggestions will be very helpful

**Comment by s-trinh** [comment]
You can find some examples here:
- https://github.com/lagadic/visp_ros/blob/aa513db52c6ba7b718f431dd01b38f0421776712/CMakeLists.txt
- https://github.com/lagadic/visp_ros/blob/aa513db52c6ba7b718f431dd01b38f0421776712/package.xml

This way, you can use ViSP as a library in your ROS2 package/node.
